File: apps/web-bff/src/main/java/com/neuron/bff/model/Budget.java
package com.neuron.bff.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Budget model representing a budget plan.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Budget {

    private String id;
    private String name;
    private String category;
    private Double limit;
    private Double spent;
    private Double remaining;
    private String period; // monthly, weekly, yearly
    private String startDate;
    private String endDate;
    private String createdAt;
    private String updatedAt;
}

File: backend/app/models/budget.py
"""Budget model."""

from datetime import datetime
from decimal import Decimal
from uuid import uuid4

from sqlmodel import Field, SQLModel


class Budget(SQLModel, table=True):
    """Budget model."""

    __tablename__ = "budgets"

    id: str = Field(primary_key=True, default_factory=lambda: str(uuid4()))
    user_id: str = Field(index=True)
    category: str
    amount: Decimal
    period: str  # weekly, monthly, yearly
    start_date: datetime
    end_date: datetime
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)
